File: src/main/java/com/ort/fw/security/UserInfoService.java
package com.ort.fw.security;

import org.dbflute.optional.OptionalEntity;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.ort.dbflute.exbhv.PlayerBhv;
import com.ort.dbflute.exentity.Player;

@Service
public class UserInfoService implements UserDetailsService {

    // ===================================================================================
    //                                                                           Attribute
    //                                                                           =========
    @Autowired
    private PlayerBhv playerBhv;

    // ===================================================================================
    //                                                                             Execute
    //                                                                             =======
    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        if (username == null || "".equals(username)) {
            throw new UsernameNotFoundException("Username is empty");
        }
        OptionalEntity<Player> optPlayer = playerBhv.selectEntity(cb -> {
            cb.query().setPlayerName_Equal(username);
        });
        if (!optPlayer.isPresent()) {
            throw new UsernameNotFoundException("User not found for name: " + username);
        }
        UserInfo userInfo = new UserInfo();
        optPlayer.ifPresent(player -> {
            userInfo.setUsername(player.getPlayerName());
            userInfo.setPassword(player.getPlayerPassword());
            userInfo.setAuthority(player.getAuthorityCodeAsAuthority());
        });

        return userInfo;
    }
}

File: backend/src/main/kotlin/com/ort/app/fw/security/UserInfoService.kt
package com.ort.app.fw.security

import com.ort.dbflute.cbean.PlayerCB
import com.ort.dbflute.exbhv.PlayerBhv
import com.ort.dbflute.exentity.Player
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserInfoService(
    private val playerBhv: PlayerBhv
) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String?): UserDetails? {
        if (username.isNullOrEmpty()) {
            throw UsernameNotFoundException("Username is empty")
        }
        val optPlayer = playerBhv.selectEntity { cb: PlayerCB ->
            cb.query().setPlayerName_Equal(username)
        }
        if (!optPlayer.isPresent) {
            throw UsernameNotFoundException("User not found for name: $username")
        }
        val userInfo = UserInfo()
        optPlayer.ifPresent { player: Player ->
            userInfo.username = player.playerName
            userInfo.password = player.playerPassword
            userInfo.setAuthority(player.authorityCodeAsAuthority)
        }
        return userInfo
    }
}